fili-core/src/main/java/com/yahoo/bard/webservice/data/config/provider/PhysicalTableConfiguration.java

```diff
@@ -0,0 +1,31 @@
+// Copyright 2016 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.data.config.provider;
+
+import com.yahoo.bard.webservice.data.config.dimension.DimensionConfig;
+import com.yahoo.bard.webservice.data.config.names.FieldName;
+import com.yahoo.bard.webservice.data.config.table.PhysicalTableDefinition;
+
+import java.util.Set;
+
+/**
+ * A PhysicalTableConfiguration can create its PhysicalTableDefinition, given a dimension configuration.
+ */
+public interface PhysicalTableConfiguration {
+
```

Comment:
Do you have any implementing code for this class? 
What's the set of assumptions about how the metrics will be provided and how the dimension map will be used?